src/p4est_communication.h

```diff
@@ -575,6 +582,184 @@ void                p4est_transfer_items_end (p4est_transfer_context_t * tc);
+ * 
+ * Points can be instances of an arbitrary struct. Point-quadrant intersection
+ * is specified by the user supplied callback \a intersect. A single point may
+ * intersect multiple process domains, and after communication will be known
+ * to each of these processes.
+ * 
+ * Each process is responsible for propagating a subset of the points it
+ * knows. Before communication, exactly one process should be responsible for
+ * propagating each point. The intersecting processes for each point are
+ * determined - by the responsible process - with \ref p4est_search_partition.
+ * Points are then communicated to the relevant processes. The algorithm
+ * ensures that after communication exactly one process is responsible for the
+ * propagation of each point. This is the process with the lowest rank among
+ * processes intersecting the point. Points known to a process before
+ * communication that do not intersect its domain are forgotten. The option
+ * \a save_unowned can be used to avoid forgetting points that do not
+ * intersect the domain of any process. If this option is enabled then these
+ * points are remembered by the process that was responsible for propagating
+ * them.
+ * 
+ * The points that a process is responsible for propagating are stored in a
+ * subarray of the array of known points, as described in 
+ * \ref p4est_transfer_search_t. Users should take care to maintain this
+ * subdivision if they modify the array of points between rounds of
+ * communication.
+ * 
+ * \param [in] p4est        The forest we search with. Its user_pointer is
+ *                          passed to the intersection callback.
+ * \param [in,out] c        Points and propagation responsibilities. The
+ *                          array \a c.points is destroyed and reallocated,
+ *                          so pointers to it should not be referenced after
+ *                          this function has been called.
+ * \param [in] intersect    Intersection callback.
+ * \param [in] save_unowned If true then points that would be unowned are
+ *                          maintained by their propagating process
+ * \return 0 if transfer was successful.
+ */
+int
+p4est_transfer_search (p4est_t *p4est, p4est_transfer_search_t *c, 
+                       p4est_intersect_t intersect, int save_unowned);
+
+/** The same as \ref p4est_transfer_search, except that we search with a
+ * partition, rather than an explicit p4est. The partition can be that of any
+ * p4est, not necessarily known to the caller.
+ * 
+ * This function is collective.
+ * 
+ * \param [in] gfq          Partition offsets to traverse.  Length \a nmemb + 1.
+ * \param [in] gfp          Partition position to traverse.  Length \a nmemb + 1.
+ * \param [in] nmemb        Number of processors encoded in \a gfq (plus one).
+ * \param [in] num_trees    Tree number must match the contents of \a gfq.
+ * \param [in] user_pointer Passed to the intersection callback.
```

Comment:
Can we be more explicit on the user_pointer being part of a dummy p4est, and which elements of the dummy p4est are meaningfully populated?

Reply:
Maybe we should define a central p4est_init_dummy function to standardize dummy use wherever it occurs.

Reply:
> Maybe we should define a central p4est_init_dummy function to standardize dummy use wherever it occurs.

This may be better done in a separate PR though.